ui/src/hooks/useFetchStatuses.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from "react"
+
+export interface StackStatus {
+  Name: string
+  Error: string
+  Revision: string
+  RepoURL: string
+}
+
+const devData: StackStatus[] = [
+  {
+    Name: "Project Alpha",
+    Error: "",
+    Revision: "v1.0.1",
+    RepoURL: "https://github.com/user/project-alpha"
+  },
+  {
+    Name: "Project Beta",
+    Error: "Failed to build",
+    Revision: "v2.3.4",
+    RepoURL: "https://github.com/user/project-beta"
+  },
+  {
+    Name: "Project Gamma",
+    Error: "",
+    Revision: "v0.9.8",
+    RepoURL: "https://github.com/user/project-gamma"
+  }
+]
+
+async function fetchData(): Promise<StackStatus[]> {
+  const response = await fetch("/stacks")
+  if (!response.ok) {
+    throw new Error("Network response was not ok")
+  }
+
+  return (await response.json()) as StackStatus[]
+}
+
+export default function useFetchStatuses(): {
+  stacks: StackStatus[]
+  error: string | null
+} {
+  const [statuses, setStatuses] = useState<StackStatus[]>([])
+  const [error, setError] = useState<string | null>(null)
+
+  useEffect(() => {
+    if (import.meta.env.MODE === "development") {
+      setStatuses(devData)
```

Comment:
Not a fan of baking development mode code into the source like this. I'd use a controller class and use inheritance to separate the two.

Reply:
I separated it into a single function call to isolate it but we are using functional programming with no inheritance here... Also conditional use of hooks is frowned upon

Reply:
Fair enough